### PR TITLE
Configuration implementation

### PR DESCRIPTION
**Pull request by darosior**
This implements the configuration of our yet to come `revaultd` binary.

Functionnalities-wise:
- Our default configuration directory is `~/.revault/` on Linux-based OSes, `Library/Application Support/revault/` on macOS, and `AppData\revault\revault.toml` on Windows (the two latters are untested, will add a CI test for them). Though as we'll accept a `-conf` argument in the future to specify the conf file and the data dir can be specified in it, it can be changed. I had to introduce a crate for the boilerplate...
- The configuration file is mandatory, as a setup is required in order to operate.
- Configuration options will need to be set in the configuration file, we disallow setting them on the command line: futile duplication.
- We use a TOML file, as it appeared to be an obvious choice, at least to me ?

Code-wise:
- src/config.rs handles the user-set, static configuration deserialization and parsing
- src/revaultd.rs handles our mutable "global state", which for now basically only contains the miniscript descriptors.

EDIT: i'm going stuck by Miniscript support for xpubs soon..

### CODE REVIEW

**Comment by darosior** [comment]
Rebased on master to get Travis.

Force pushed to:
- Lower the serde version (somehow this fixed the Win build)
- Fix the Window test (`s/AppData/AppData\/Roaming/`)

**Comment by darosior** [comment]
Renewed the Cargo.lock ..

**Comment by darosior** [comment]
Removed the nightly + windows build...

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+use std::vec::Vec;
+
+use bitcoin::util::bip32::ExtendedPubKey;
+use serde::{de, Deserialize, Deserializer};
+
+/// Everything we need to know for talking to bitcoind serenely
+#[derive(Debug, Deserialize, Clone)]
+pub struct BitcoindConfig {
+    /// The network we are operating on, one of "mainnet", "testnet", "regtest"
+    pub network: String,
```
Comment (edouardparis) [inline src/config.rs:14]: I like to order fields alphabetically, but ok for me

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    // TODO: think more about our potential need for the datadir
+    pub cookie_path: PathBuf,
+    /// The IP:port bitcoind's RPC is listening on
+    pub addr: SocketAddr,
+}
+
+/// A participant not taking part in day-to-day fund management, and who runs
+/// a cosigning server to ensure that spending transactions are only signed once.
+#[derive(Debug)]
+pub struct NonManager {
```
Comment (edouardparis) [inline src/config.rs:25]: I am not sure I like the name, maybe `Spender`

Comment (darosior) [reply src/config.rs:25]: Some rationale on https://github.com/re-vault/practical-revault/pull/13 and on MM: https://chat.revault.dev:9876/revault/pl/r6twixf65p8f9d5k1r6wpxjf7r

**Comment by darosior** [comment]
self-ACK 8d18164

**Review by JSwambo — COMMENTED** [review]
Ack!

**Pull request merged**